src/canvas/Document.tsx

```diff
@@ -28,4 +30,22 @@ export class Document implements IDocument {
         }
         return Boolean(this._patternPaths.pop());
     }
+
+    // Sets the pixels per inch ratio according to the input measurement
+    setSizeRatio = (inputMeasurementInInches: number): void => {
+        this._patternPaths.forEach(path => {
+            if (path.isSelected()) {
```

Comment:
I think we should move selection to its own class, which holds a reference to the path that is selected. It doesn't make sense to put it on the path because it is temporary state. Additionally, it is more convenient to check which path is selected. Also, it is more flexible in case we want to dispatch events in the future when the selected path changes.

Reply:
I agree it's clunky and I did code it differently at first, but I wanted to keep as simple as possible in _drawPatternPaths in the renderer, because that gets called many times per second. how would your solution work there?

Reply:
If we hold a reference in Selection, we could do something like
if(path === selection.selected)

Reply:
ok, I tried something out and pushed it. I led to some weird stuff when accessing the selected path in addmeasurement, so I neeeded yet another getter in the renderer to access that. Let me know what you think.